app/api/auth/oauth.py

```diff
@@ -0,0 +1,78 @@
+# app/auth/oauth.py
+
+from uuid import UUID
+from dotenv import load_dotenv
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+import os
+import jwt
+from app.models import TokenData
+from app.models import User
+from app.api.auth.auth import fake_db_user
+
+load_dotenv()
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/login/")
+
+async def get_current_user(token: str = Depends(oauth2_scheme)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, os.getenv("SECRET_KEY"), algorithms=[os.getenv("ALGORITHM")])
+        user_id: str = payload.get("sub")
+        if user_id is None:
+            raise credentials_exception
+    except jwt.InvalidTokenError:
+        raise credentials_exception
+    user = next((u for u in fake_db_user.values() if str(u["id"]) == user_id), None) 
+    if user is None:
+        raise credentials_exception
+    return TokenData(user_id=UUID(user_id))
```

Comment:
this is called get_current_user why does it return Token, it should return the current user